### DISCUSSION TITLE
Versioning/pinning of sidecar image

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#21195] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Every time Renovate runs something in a sidecar container, it does `docker pull ..X../sidecar` which hits Docker Hub or GHCR. I would instead prefer to use versions of sidecar so that it can more efficiently use a pull-through cache and not touch GHCR or DH. It would also be good practice to version the sidecar.

I can think of a few options:
1. Don't version sidecar in Renovate itself, but allow the version to be configured (default: latest). Then it can be controlled/updated by e.g. someone's docker composer file
2. Version it in Renovate, as default value in a config option. Use regex manager to update
3. Version it in renovate somewhere in lib/util/exec, non-configurable. Use regex manager to update

**Comment by secustor** [comment]
I would be in favour for option 2, which would also allow to define fully custom images to be used as sidecars.

**Reply by viceice** [reply]
agreed

**Reply by rarkins** [reply]
So a new global config option, which is a fully qualified image/tag?

**Reply by viceice** [reply]
yes and remove/ migrate docker image prefix.

**Reply by rarkins** [reply]
Ok in v36 then

**Comment by rarkins** [comment]
Created feature request: #21217
